src/helpers.rs

```diff
@@ -41,11 +55,13 @@ pub fn read_file(path: &str) -> io::Result<String> {
 }
 
 /// Indent every line in text by four spaces.
+#[cfg(any(feature = "html_root_url", feature = "markdown"))]
```

Comment:
This is an example of a conditional which seems strange at first — this little innocent function doesn't depend on anything outside of `std`, so surely we should be able to just define it? I guess no since we'll get warnings about unused functions and so... but perhaps we could suppress them instead.

I'm not sure it's a super idea, but perhaps we could allow unused imports and functions with `#![allow(dead_code)]`, but then in CI run a step with `RUSTFLAGS='-D dead_code' cargo build`. However, from a little testing just now, it doesn't seem to work the way I hoped.

Reply:
I moved most of the imports to the functions using them. The remaining ones in `helpers.rs` are in the function signatures, and the cfg annotations are required for those as they become undefined imports then that dependency is disabled.

Reply:
Resolving this one, but feel free to make further comments about this if you want.